src/connectors/openml.py

```diff
@@ -32,7 +45,14 @@ def fetch_dataset(dataset: Dataset) -> dict:
         "name": dataset_json["name"],
         "description": dataset_json["description"],
         "file_url": dataset_json["url"],
-        "number_of_samples": qualities_json["NumberOfInstances"],
-        "number_of_features": qualities_json["NumberOfFeatures"],
-        "number_of_classes": qualities_json.get("NumberOfClasses"),
+        "number_of_samples": _as_int(qualities_json["NumberOfInstances"]),
+        "number_of_features": _as_int(qualities_json["NumberOfFeatures"]),
+        "number_of_classes": _as_int(qualities_json["NumberOfClasses"]),
```

Comment:
I vaguely remembered some `NumberOfClasses` being absent for regression datasets (which is why I used `get`), but I can't find an example. Perhaps the server is updated. If you want to be sure, you can write a quick script that downloads qualities for all the datasets to verify they all contain these keys.